### PR TITLE
feat(ember-flight-icons): convert to typescript & publish types

### PR DESCRIPTION
**Pull request by aklkv**
### :pushpin: Summary

The main goal of this change is to make sure types are created and published for the Ember Flight Icon component

### :link: External links

Jira ticket: https://hashicorp.atlassian.net/browse/HDS-3315

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/ember-flight-icons/src/components.ts
Comment (aklkv) [inline packages/ember-flight-icons/src/components.ts:3]: this gives consumers the ability to import components like so:

```js
import { FlightIcon } from '@hashicorp/ember-flight-icons/components'
```

File: packages/components/src/components/hds/app-footer/link.ts
Comment (didoo) [inline packages/components/src/components/hds/app-footer/link.ts:11]: Question: should we instead import the signature from the `EmberFlightIcon` component, and then use its `FlightIconSignature['Args']['name']?

File: .changeset/long-rats-share.md
Comment (didoo) [inline .changeset/long-rats-share.md:5]: This needs to be updated

File: packages/components/package.json
Comment (didoo) [inline packages/components/package.json:35]: In the other `package.json` files we have only the `start` command. Are the other two needed? What are they used for? Doesn't the `start` command do the same (only a bit more)?
```suggestion
    "start": "rollup --config --watch --environment development",
```

File: packages/components/unpublished-development-types/@hashicorp/ember-flight-icons/flight-icon.d.ts
Comment (didoo) [inline packages/components/unpublished-development-types/@hashicorp/ember-flight-icons/flight-icon.d.ts:1]: Question: I see that this file has been deleted, but could it have been an option to just adde the `IconName` signature here, without converting the `EmberFlightIcon` component to TS? In that case, can you follow this minimum path with the PR? As mentioned by @alex-ju, we're planning to create a brand new component `Hds::Icon` within the `components` package (relatively soon) so it would be better to have the simplest PR possible here, much easier to review and approve.

**Review by didoo — COMMENTED** [review]
@aklkv left a few comments

File: packages/components/package.json
Comment (aklkv) [reply packages/components/package.json:35]: I guess it just a convenience command that got just introduced in the latest blueprint, happy drop this changes is there is no value

Comment (aklkv) [reply packages/components/package.json:35]: [components](https://github.com/hashicorp/design-system/blob/a3c3f3da5abea4fecdce9d0d190562aa9241664c/packages/components/package.json#L34) packages has similar commands but they are bypassed for some reason, happy to drop this change if you see no value

File: packages/ember-flight-icons/babel.config.json
Comment (alex-ju) [inline packages/ember-flight-icons/babel.config.json:7]: unfortunately, we can't remove these – they are still needed for older codebases (such as `atlas`) to work. I ended up re-adding them in `components`.

File: packages/ember-flight-icons/package.json
Comment (alex-ju) [inline packages/ember-flight-icons/package.json:39]: I believe we use a more recent version of `decorator-transforms` in `components`; we can try to align them

**Review by alex-ju — COMMENTED** [review]
[CI doesn't seem to be happy with the new setup](https://github.com/hashicorp/design-system/actions/runs/9350118677/job/25732918986?pr=2126) (although it appears to be building fine locally) – would you be keen to investigate it?

**Comment by alex-ju** [comment]
> @alex-ju I definitely fix that, I might be a little slower since I am traveling for the next couple of weeks

no rush on our side – safe travels! ✈️

File: packages/ember-flight-icons/src/components/flight-icon.hbs
Comment (didoo) [inline packages/ember-flight-icons/src/components/flight-icon.hbs:1]: I think this header needs to be preserved

File: packages/ember-flight-icons/src/components/flight-icon.ts
Comment (didoo) [inline packages/ember-flight-icons/src/components/flight-icon.ts:17]: Currently we don't have inline comments to describe the arguments in the other signatures. Do you mind removing them? We may decide later to do it systematically across all the codebase, but for now I prefer to have consistency in how we declare the signatures, to avoid having a jungle of formats. Thank you

File: packages/ember-flight-icons/src/components/flight-icon.ts
Comment (didoo) [inline packages/ember-flight-icons/src/components/flight-icon.ts:22]: Should this be required? 🤔

File: packages/ember-flight-icons/src/components/flight-icon.ts
Comment (didoo) [inline packages/ember-flight-icons/src/components/flight-icon.ts]: why this extra comma at the end of the line?
modified

File: packages/ember-flight-icons/src/components/flight-icon.ts
Comment (didoo) [inline packages/ember-flight-icons/src/components/flight-icon.ts:14]: We [tend to follow](https://github.com/hashicorp/design-system/blob/main/wiki/TypeScript-Migration.md#typescript-migration) the same order as in the documentation, so it's easy to review and compare:
https://helios.hashicorp.design/icons/usage-guidelines?tab=code#component-api

File: packages/ember-flight-icons/package.json
Comment (didoo) [inline packages/ember-flight-icons/package.json]: Should these changes go in https://github.com/hashicorp/design-system/pull/2142 ?

File: packages/ember-flight-icons/package.json
Comment (didoo) [inline packages/ember-flight-icons/package.json:86]: what is this? 🤔

**Review by didoo — COMMENTED** [review]
Left a few comments (after rebasing on https://github.com/hashicorp/design-system/pull/2142/)

File: packages/components/package.json
Comment (didoo) [reply packages/components/package.json:35]: @aklkv we discussed that these changes would be removed or moved to https://github.com/hashicorp/design-system/pull/2142. Did I misunderstand?

File: packages/components/unpublished-development-types/@hashicorp/ember-flight-icons/flight-icon.d.ts
Comment (didoo) [reply packages/components/unpublished-development-types/@hashicorp/ember-flight-icons/flight-icon.d.ts:1]: Discussed in person with @aklkv and clarified that just changing this file would not emit the types (in fact, the file lives under the `unpublished-development-types` folder)

File: packages/ember-flight-icons/src/components/flight-icon.ts
Comment (aklkv) [reply packages/ember-flight-icons/src/components/flight-icon.ts]: in the newer prettier version dangling comma is a default now

File: packages/ember-flight-icons/package.json
Comment (aklkv) [reply packages/ember-flight-icons/package.json]: missconfugured rollup

File: packages/ember-flight-icons/src/components/flight-icon.ts
Comment (didoo) [reply packages/ember-flight-icons/src/components/flight-icon.ts]: left a comment about it. I would prefer if updates like this are done systematically across the monorepo, to avoid that different parts behave in different ways.

File: packages/ember-flight-icons/src/components/flight-icon.ts
Comment (aklkv) [reply packages/ember-flight-icons/src/components/flight-icon.ts:15]: I have tried that but, several consuming components are not happy about this change, I could potentially look into fixing the consuming side but since `Hds::Icon` is coming pretty soon I would say we have an assert segment that would catch potential errors, this should be enough till the new `Hds::Icon` is shipped

Comment (didoo) [reply packages/ember-flight-icons/src/components/flight-icon.ts:15]: OK, makes sense. 👍

**Review by didoo — APPROVED** [review]
I _think_ is OK, but let's wait for @alex-ju to approve too before merging

File: .changeset/long-rats-share.md
Comment (alex-ju) [inline .changeset/long-rats-share.md:5]: nit (we have some very specific guidance about [consistent terminology](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#consistent-terminology) for the changelog)
```suggestion
`FlightIcon` - converted component to TypeScript and published types
```

**Review by alex-ju — APPROVED** [review]
Looking great! Thanks for addressing all the feedback

**Pull request merged**
